### ISSUE TITLE
setUniform not working

### ISSUE THREAD

**Issue by RainCatalyst** [patriciogonzalezvivo/glslCanvas#67]
Here is a minimal reproducible example of the issue. I should get just a gray canvas but it looks like u_gray uniform is always zero. Am I doing something wrong?

_index.js_
```js
var canvas = document.getElementById('glslCanvas');
var sandbox = new GlslCanvas(canvas);
sandbox.setUniform("u_gray", 0.5);
```
_index.html_
```html
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <title>Test</title>
	<script type="text/javascript" src="https://rawgit.com/patriciogonzalezvivo/glslCanvas/master/dist/GlslCanvas.js"></script>
    </head>
    <body>
	<canvas id="glslCanvas" class='glslCanvas' data-fragment-url="shader.frag" width="512" height="512"></canvas>
	<script type="text/javascript" src="index.js"></script>
    </body>
</html>
```
_shader.frag_
```glsl
precision mediump float;

uniform float u_gray;

void main() {
  gl_FragColor = vec4(vec3(u_gray), 1.0);
}
```

**Comment by 2nafish117** [comment]
Yep, have the same issue. The js console says `Uncaught TypeError: Cannot read property 'useProgram' of undefined
    at GlslCanvas.uniform (GlslCanvas.js:1769)
    at GlslCanvas.setUniforms (GlslCanvas.js:1729)
    at GlslCanvas.setUniform (GlslCanvas.js:1716)`

**Comment by myreauks** [comment]
I have the same issue with setUniform, although I do not get errors. The value just does no transfre to the shader.
Here is my code:
`        var canvas = document.getElementById("glslCanvas");
        
        if (canvas) {
            const sandbox = new GlslCanvas(canvas);
            let hue = 1.5;
            sandbox.setUniform('u_hue', hue);
        }`

**Comment by bnlcas** [comment]
I'm having the same issue. The GlslCanvas is not correctly updating the list of uniforms for me (Chrome 106 - caching disabled).
Will report any progress on this.

**Comment by thisisseb** [comment]
I having the same issue. As a workaround, I noticed that setting uniforms works fine when loading the fragment manually using `sandbox.load()`, with the call to the `load` method before the call(s) to `setUniform`.

So for now to get around the issue, I'm doing something like this, which seems to work well for me:

```
const canvas = document.querySelector('#my-canvas');
const sandbox = new GlslCanvas(canvas);

document.addEventListener('DOMContentLoaded', () => {
  fetch(`/path/to/shader.frag`)
    .then(res => res.text())
    .then(fragmentText => {
      sandbox.load(fragmentText);
      sandbox.setUniform('u_brightness', 0.5);
    });
});
```
